Source/WebCore/dom/make_names.pl

```diff
@@ -167,75 +169,126 @@ END
+if ($printEnum eq "") {
+    $parameters{fallbackJSInterfaceName} = $parameters{fallbackInterfaceName} unless $parameters{fallbackJSInterfaceName};
+}
 
-$parameters{namespacePrefix} = $parameters{namespace} unless $parameters{namespacePrefix};
-$parameters{fallbackJSInterfaceName} = $parameters{fallbackInterfaceName} unless $parameters{fallbackJSInterfaceName};
+collectAllElementsAndAttrsPerNamespace();
 
-my $typeHelpersBasePath = "$outputDir/$parameters{namespace}ElementTypeHelpers";
-my $namesBasePath = "$outputDir/$parameters{namespace}Names";
-my $factoryBasePath = "$outputDir/$parameters{namespace}ElementFactory";
-my $wrapperFactoryFileName = "$parameters{namespace}ElementWrapperFactory";
+if ($printEnum eq "") {
+    my $typeHelpersBasePath = "$outputDir/$parameters{namespace}ElementTypeHelpers";
+    my $namesBasePath = "$outputDir/$parameters{namespace}Names";
 
-printNamesHeaderFile("$namesBasePath.h");
-printNamesCppFile("$namesBasePath.cpp");
-printTypeHelpersHeaderFile("$typeHelpersBasePath.h");
+    printNamesHeaderFile("$namesBasePath.h");
+    printNamesCppFile("$namesBasePath.cpp");
+    printTypeHelpersHeaderFile("$typeHelpersBasePath.h");
+}
 
 if ($printFactory) {
+    my $factoryBasePath = "$outputDir/$parameters{namespace}ElementFactory";
+
     printFactoryCppFile("$factoryBasePath.cpp");
     printFactoryHeaderFile("$factoryBasePath.h");
 }
 
 if ($printWrapperFactory) {
+    my $wrapperFactoryFileName = "$parameters{namespace}ElementWrapperFactory";
+
     printWrapperFactoryCppFile($outputDir, $wrapperFactoryFileName);
     printWrapperFactoryHeaderFile($outputDir, $wrapperFactoryFileName);
 }
 
+if ($printEnum eq "TagName") {
+    printTagNameHeaderFile("$outputDir/TagName.h");
+    printTagNameCppFile("$outputDir/TagName.cpp");
+}
+
+if ($printEnum eq "ElementName") {
+    printElementNameHeaderFile("$outputDir/ElementName.h");
+    printElementNameCppFile("$outputDir/ElementName.cpp");
+}
+
+if ($printEnum eq "Namespace") {
+    printNamespaceHeaderFile("$outputDir/Namespace.h");
+    printNamespaceCppFile("$outputDir/Namespace.cpp");
+}
+
 ### Hash initialization
 
-sub defaultTagPropertyHash
+sub defaultElementPropertyHash
 {
+    my $localName = shift;
+
+    my $identifier = $localName =~ s/-/_/gr;
+    my $requiresAdjustment = $localName ne lc $localName;
+    my $tagEnumValue = avoidConflictingName($identifier);
+
     return (
-        'constructorNeedsCreatedByParser' => 0,
-        'constructorNeedsFormElement' => 0,
-        'noConstructor' => 0,
-        'interfaceName' => defaultInterfaceName($_[0]),
+        cppNamespace => $parameters{namespace},
+        namespace => $parameters{namespace},
+        localName => $localName,
+        identifier => $identifier,
+        tagEnumValue => $tagEnumValue,
+        elementEnumValue => "$parameters{namespace}_$identifier",
+        unadjustedTagEnumValue => $requiresAdjustment ? "Unadjusted_" . lc($identifier) : "",
```

Comment:
As we discussed before, we should probably use a more descriptive prefix like UnadjectedCase_.
I think suffix like altglyphCaseUnadjusted would read best.

Reply:
Ah yes, I forgot.